gcloud-java-compute/src/main/java/com/google/gcloud/compute/RegionForwardingRuleId.java

```diff
@@ -0,0 +1,154 @@
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Identity for a Google Compute Engine region's forwarding rule.
+ */
+public final class RegionForwardingRuleId extends RegionResourceId implements ForwardingRuleId {
+
+  static final Function<String, RegionForwardingRuleId> FROM_URL_FUNCTION =
+      new Function<String, RegionForwardingRuleId>() {
+        @Override
+        public RegionForwardingRuleId apply(String pb) {
+          return RegionForwardingRuleId.fromUrl(pb);
+        }
+      };
+  static final Function<RegionForwardingRuleId, String> TO_URL_FUNCTION =
+      new Function<RegionForwardingRuleId, String>() {
+        @Override
+        public String apply(RegionForwardingRuleId forwardingRuleId) {
+          return forwardingRuleId.selfLink();
+        }
+      };
+
+  private static final String REGEX = RegionResourceId.REGEX + "forwardingRules/([^/]+)";
+  private static final Pattern PATTERN = Pattern.compile(REGEX);
+  private static final long serialVersionUID = 7885327931402904667L;
+
+  private final String rule;
+
+  private RegionForwardingRuleId(String project, String region, String rule) {
+    super(project, region);
+    this.rule = checkNotNull(rule);
+  }
+
+  @Override
+  public Type type() {
+    return Type.REGION;
+  }
+
+  @Override
+  public String rule() {
+    return rule;
+  }
+
+  @Override
+  public String selfLink() {
+    return super.selfLink() + "/forwardingRules/" + rule;
+  }
+
+  @Override
+  MoreObjects.ToStringHelper toStringHelper() {
+    return MoreObjects.toStringHelper(this).add("rule", rule);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(baseHashCode(), rule);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof RegionForwardingRuleId
+        && baseEquals((RegionForwardingRuleId) obj)
+        && Objects.equals(rule, ((RegionForwardingRuleId) obj).rule);
+  }
+
+  @Override
+  RegionForwardingRuleId setProjectId(String projectId) {
+    if (project() != null) {
+      return this;
+    }
+    return RegionForwardingRuleId.of(projectId, region(), rule);
+  }
+
+  /**
+   * Returns a region forwarding rule identity given the region identity and the rule name. The
+   * forwarding rule name must be 1-63 characters long, and comply with RFC1035. Specifically, the
+   * name must be 1-63 characters long and match the regular expression
+   * {@code [a-z]([-a-z0-9]*[a-z0-9])?} which means the first character must be a lowercase letter,
+   * and all following characters must be a dash, lowercase letter, or digit, except the last
+   * character, which cannot be a dash.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  public static RegionForwardingRuleId of(RegionId regionId, String operation) {
```

Comment:
Should operation --> rule, or is a rule a type of operation?  (question applies for the other `of` methods in this class as well)